### PR TITLE
feat: [XD-6] UI package

### PR DESCRIPTION
**Pull request by knagurski**
This PR adds the UI package to the repo and moves the `Alert` component to the package.

### CODE REVIEW

File: packages/playground/src/components/image-carousel.tsx [hunk 1]
```diff
@@ -51,28 +52,26 @@ const ImageCarousel = (props: ImageCarouselProps) => {
...
-                  tooltip={'Zoom in'}
                   onClick={() => {
                     if (Number(zoomLevel.toFixed(1)) < 2) {
                       setZoomLevel(zoomLevel + ZOOM_INC_DEC_LEVEL)
                     }
                   }}
-                />
+                  title="Zoom in">
+                  +
+                </Button>
```
Comment (knagurski) [inline packages/playground/src/components/image-carousel.tsx:63]: The `icon` variant doesn't exist. Nor does the `icon` or `tooltip` props.

File: packages/playground/src/components/image-carousel.tsx [hunk 2]
```diff
@@ -51,28 +52,26 @@ const ImageCarousel = (props: ImageCarouselProps) => {
             <div>
               <div className="flex">
                 <Button
-                  variant={'icon'}
-                  icon="zoom-in"
                   data-testid="zoomInButton"
-                  tooltip={'Zoom in'}
```
Comment (knagurski) [inline packages/playground/src/components/image-carousel.tsx:57]: The `icon` variant doesn't exist. Nor does the `icon` or `tooltip` props.

File: packages/playground/src/components/image-carousel.tsx [hunk 3]
```diff
@@ -51,28 +52,26 @@ const ImageCarousel = (props: ImageCarouselProps) => {
...
-                />
+                  title="Zoom in">
+                  +
+                </Button>
 
                 <Button
-                  variation={'icon'}
-                  icon="zoom-out"
                   data-testid="zoomOutButton"
-                  tooltip={'Zoom out'}
```
Comment (knagurski) [inline packages/playground/src/components/image-carousel.tsx:69]: As above

File: packages/playground/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx [hunk 4]
```diff
@@ -225,7 +224,7 @@ export const BranchSettingsRuleBypassListField: React.FC<FieldProps & { bypassOp
       <DropdownMenu>
         <DropdownMenuTrigger>
           <div className="flex items-center justify-between rounded-md border">
-            <Button variant="ghost w-full">
```
Comment (knagurski) [inline packages/playground/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx:228]: Invalid variant. Fixed to be variant and class

File: packages/playground/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx [hunk 5]
```diff
@@ -342,7 +341,7 @@ export const BranchSettingsRuleListField: React.FC<{
               <DropdownMenu>
                 <DropdownMenuTrigger className="w-full">
                   <div className="flex items-center justify-between rounded-md border">
-                    <Button variant="ghost w-full">
```
Comment (knagurski) [inline packages/playground/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx:345]: As above

File: packages/playground/src/components/user-management/form-admin-remove-dialog.tsx [hunk 6]
```diff
@@ -30,7 +30,7 @@ export const FormRemoveAdminDialog: React.FC<FormRemoveUserDialogProps> = ({
         <AlertDialogHeader>
           <AlertDialogTitle>
             Are you sure you want to remove
-            <Badge type="admin" className="mx-2" variant="muted" disableHover={true}>
```
Comment (knagurski) [inline packages/playground/src/components/user-management/form-admin-remove-dialog.tsx:33]: Invalid variant and prop

File: packages/playground/src/components/user-management/form-user-reset-password.tsx [hunk 7]
```diff
@@ -38,7 +38,7 @@ export const FormResetPasswordDialog: React.FC<FormResetPasswordsDialogProps> =
           ) : (
             <AlertDialogTitle>
               Are you sure you want to reset password for
-              <Badge type="admin" className="mx-2" variant="muted" disableHover={true}>
```
Comment (knagurski) [inline packages/playground/src/components/user-management/form-user-reset-password.tsx:41]: Invalid variant and prop

File: packages/playground/src/components/user-management/form-user-reset-password.tsx [hunk 8]
```diff
@@ -66,7 +66,7 @@ export const FormResetPasswordDialog: React.FC<FormResetPasswordsDialogProps> =
           {!isConfirm && (
             <Button
               size="default"
-              theme="secondary"
```
Comment (knagurski) [inline packages/playground/src/components/user-management/form-user-reset-password.tsx:69]: `theme` doesn't exist on `Button`

File: packages/playground/tsconfig.json [hunk 9]
```diff
@@ -8,7 +8,7 @@
     "esModuleInterop": false,
     "allowSyntheticDefaultImports": true,
     "strict": true,
-    "moduleResolution": "Node",
+    "moduleResolution": "Bundler",
```
Comment (knagurski) [inline packages/playground/tsconfig.json:11]: Standardising resolution

File: packages/ui/vite.config.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      provider: 'istanbul',
+      include: ['src'],
+      exclude: ['src/index.ts', 'src/components/index.ts', 'src/views/index.ts', 'src/**/*.test.*', 'src/utils/cn.ts'],
+      extension: ['ts', 'js', 'tsx', 'jsx']
+      // thresholds: {
+      //   branches: 80,
+      //   lines: 80,
+      //   functions: 80,
+      //   statements: 80
+      // }
```
Comment (knagurski) [inline packages/ui/vite.config.ts:40]: To be enabled in the future

File: packages/views/package.json [hunk 11]
```diff
@@ -3,12 +3,10 @@
...
   "description": "Harness Unified Pipeline UI views",
   "scripts": {
-    "dev": "run-p watch:css watch",
-    "build:css": "npx tailwindcss -i ./src/styles.css -o ./public/styles.css",
-    "watch:css": "pnpm build:css --watch",
+    "dev": "run-p watch",
     "preinstall": "npx only-allow pnpm",
     "watch": "vite",
-    "build": "pnpm build:css && vite build",
+    "build": "vite build",
```
Comment (knagurski) [inline packages/views/package.json:9]: Eliminated need for separate CSS build step. Now handled by Vite

File: packages/views/src/components/filter.tsx [hunk 12]
```diff
@@ -1,12 +1,11 @@
 import { useEffect, useState } from 'react'
 import { isEmpty } from 'lodash-es'
 import { ListActions, SearchBox } from '@harnessio/canary'
-import { DropdownItemProps } from '@harnessio/canary/dist/components/list-actions'
```
Comment (knagurski) [inline packages/views/src/components/filter.tsx:4]: Invalid import

File: packages/views/src/components/form-inputs/common/InputLabel.tsx [hunk 13]
```diff
@@ -17,7 +17,7 @@ function InputLabel(props: InputLabelProps): JSX.Element {
       {description && (
         <Tooltip>
           <TooltipTrigger asChild>
-            <Button size="sm" variant="icon" className="ml-2 h-5 cursor-default p-1">
```
Comment (knagurski) [inline packages/views/src/components/form-inputs/common/InputLabel.tsx:20]: `icon` variant doesn't exist

File: packages/views/src/components/image-carousel.tsx [hunk 14]
```diff
@@ -51,28 +52,26 @@ const ImageCarousel = (props: ImageCarouselProps) => {
             <div>
               <div className="flex">
                 <Button
-                  variant={'icon'}
-                  icon="zoom-in"
                   data-testid="zoomInButton"
-                  tooltip={'Zoom in'}
```
Comment (knagurski) [inline packages/views/src/components/image-carousel.tsx:57]: The `icon` variant doesn't exist. Nor does the `icon` or `tooltip` props.

File: packages/views/src/components/image-carousel.tsx [hunk 15]
```diff
@@ -51,28 +52,26 @@ const ImageCarousel = (props: ImageCarouselProps) => {
...
-                />
+                  title="Zoom in">
+                  +
+                </Button>
 
                 <Button
-                  variation={'icon'}
-                  icon="zoom-out"
                   data-testid="zoomOutButton"
-                  tooltip={'Zoom out'}
```
Comment (knagurski) [inline packages/views/src/components/image-carousel.tsx:69]: As above

File: packages/views/src/components/project-settings/form-member-delete-dialog.tsx [hunk 16]
```diff
@@ -29,7 +29,7 @@ export const FormDeleteMemberDialog: React.FC<FormDeleteMemberDialogProps> = ({
         <AlertDialogHeader>
           <AlertDialogTitle className="text-left">
             Are you absolutely sure you want to delete
-            <Badge type="admin" className="mx-2" variant="muted" disableHover={true}>
```
Comment (knagurski) [inline packages/views/src/components/project-settings/form-member-delete-dialog.tsx:32]: Invalid variant and prop

File: packages/views/src/components/pull-request/pull-request-conversation-header.tsx [hunk 17]
```diff
@@ -73,7 +73,7 @@ export const PullRequestHeader: React.FC<PullRequestTitleProps> = ({
       <div className="">
         <div className="text-tertiary-background flex space-x-2">
           <div className="flex items-center gap-2.5 text-center align-middle">
-            <Badge disableHover borderRadius="full" theme={stateObject.theme} className={`select-none justify-center`}>
```
Comment (knagurski) [inline packages/views/src/components/pull-request/pull-request-conversation-header.tsx:76]: `theme` doesn't exist

File: packages/views/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx [hunk 18]
```diff
@@ -225,7 +224,7 @@ export const BranchSettingsRuleBypassListField: React.FC<FieldProps & { bypassOp
       <DropdownMenu>
         <DropdownMenuTrigger>
           <div className="flex items-center justify-between rounded-md border">
-            <Button variant="ghost w-full">
```
Comment (knagurski) [inline packages/views/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx:228]: Invalid variant, split into variant and class

File: packages/views/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx [hunk 19]
```diff
@@ -342,7 +341,7 @@ export const BranchSettingsRuleListField: React.FC<{
               <DropdownMenu>
                 <DropdownMenuTrigger className="w-full">
                   <div className="flex items-center justify-between rounded-md border">
-                    <Button variant="ghost w-full">
```
Comment (knagurski) [inline packages/views/src/components/repo-settings/repo-branch-settings-rules/repo-branch-settings-rules-fields.tsx:345]: As above

File: packages/views/src/components/user-management/form-admin-add-dialog.tsx [hunk 20]
```diff
@@ -30,7 +30,7 @@ export const FormAddAdminDialog: React.FC<FormRemoveUserDialogProps> = ({
         <AlertDialogHeader>
           <AlertDialogTitle>
             Are you sure you want to add
-            <Badge type="admin" className="mx-2" variant="muted" disableHover={true}>
```
Comment (knagurski) [inline packages/views/src/components/user-management/form-admin-add-dialog.tsx:33]: Invalid variant and prop

File: packages/views/src/components/user-management/form-admin-remove-dialog.tsx [hunk 21]
```diff
@@ -30,7 +30,7 @@ export const FormRemoveAdminDialog: React.FC<FormRemoveUserDialogProps> = ({
         <AlertDialogHeader>
           <AlertDialogTitle>
             Are you sure you want to remove
-            <Badge type="admin" className="mx-2" variant="muted" disableHover={true}>
```
Comment (knagurski) [inline packages/views/src/components/user-management/form-admin-remove-dialog.tsx:33]: Invalid variant and prop

File: packages/views/src/components/user-management/form-user-reset-password.tsx [hunk 22]
```diff
@@ -38,7 +38,7 @@ export const FormResetPasswordDialog: React.FC<FormResetPasswordsDialogProps> =
           ) : (
             <AlertDialogTitle>
               Are you sure you want to reset password for
-              <Badge type="admin" className="mx-2" variant="muted" disableHover={true}>
```
Comment (knagurski) [inline packages/views/src/components/user-management/form-user-reset-password.tsx:41]: Invalid variant and prop

File: packages/views/src/components/user-management/form-user-reset-password.tsx [hunk 23]
```diff
@@ -66,7 +66,7 @@ export const FormResetPasswordDialog: React.FC<FormResetPasswordsDialogProps> =
           {!isConfirm && (
             <Button
               size="default"
-              theme="secondary"
```
Comment (knagurski) [inline packages/views/src/components/user-management/form-user-reset-password.tsx:69]: `theme` doesn't exist

File: packages/views/tsconfig.json [hunk 24]
```diff
@@ -8,7 +8,7 @@
     "esModuleInterop": false,
     "allowSyntheticDefaultImports": true,
     "strict": true,
-    "moduleResolution": "Node",
+    "moduleResolution": "Bundler",
```
Comment (knagurski) [inline packages/views/tsconfig.json:11]: Standardising resolution

File: packages/views/tsconfig.node.json [hunk 25]
```diff
@@ -2,8 +2,7 @@
   "compilerOptions": {
     "composite": true,
     "skipLibCheck": true,
-    "module": "ESNext",
-    "moduleResolution": "bundler",
+    "moduleResolution": "Bundler",
```
Comment (knagurski) [inline packages/views/tsconfig.node.json:5]: Standardising resolution

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**